pkg/virt-controller/watch/vmi.go

```diff
@@ -1271,14 +1272,14 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 	}
 
 	if !isTempPod(pod) && isPodReady(pod) {
-		newAnnotations := network.GeneratePodAnnotations(vmi.Spec.Networks, vmi.Spec.Domain.Devices.Interfaces, pod.Annotations[networkv1.NetworkStatusAnnot], c.clusterConfig.GetNetworkBindings())
-		if len(newAnnotations) != 0 {
-			patchedPod, err := c.syncPodAnnotations(pod, newAnnotations)
-			if err != nil {
-				return &syncErrorImpl{err, FailedPodPatchReason}
-			}
-			*pod = *patchedPod
+		newAnnotations := map[string]string{virtv1.DeschedulerRequestEvictOnlyAnnotation: ""}
+		maps.Copy(newAnnotations, network.GeneratePodAnnotations(vmi.Spec.Networks, vmi.Spec.Domain.Devices.Interfaces, pod.Annotations[networkv1.NetworkStatusAnnot], c.clusterConfig.GetNetworkBindings()))
+
+		patchedPod, err := c.syncPodAnnotations(pod, newAnnotations)
+		if err != nil {
+			return &syncErrorImpl{err, FailedPodPatchReason}
 		}
+		*pod = *patchedPod
 
 		hotplugVolumes := getHotplugVolumes(vmi, pod)
 		hotplugAttachmentPods, err := controller.AttachmentPods(pod, c.podIndexer)
```

Comment:
@fossedihelm Shouldn't we handle the AttachmentPods as well?

also in migration: https://github.com/kubevirt/kubevirt/blob/841745da53a970eed302dccd12fdcce6f98a2edb/pkg/virt-controller/watch/migration.go#L412

Reply:
attachment pods need special handling that will be covered in a separate PR. Thanks for raising this! :+1:

Reply:
Great, can we open an issue regarding this to make sure it is not forgotten?